File: python/unidad_02/problema_2_1.py
'''
El programa, dado como dato el número de sonidos emitidos por un
grillo en un minuto, calcula la temperatura en grados Fahrenheit.
'''

def calcular_temperatura(no_sonidos):

    return no_sonidos / 4 + 40
    # Función de temperatura en grados Fahrenheit
    # de acuerdo al no. de sonidos emitidos por un grillo en un minuto

def main():

    try:

        no_sonidos = int(input("Sonidos por minuto de un grillo: "))

        if no_sonidos <= 0:

            print("Error: el no. de sonidos debe ser mayor que cero.")
            return
    
    except ValueError:

        print("Error: por favor, ingresa un valor válido.")
        return
    
    temperatura = calcular_temperatura(no_sonidos)

    print("La temperatura es igual a")
    
    print(f"{temperatura:.2f} grados Fahrenheit")
    
if __name__ == '__main__':

    main()

File: java/unidad_02/Problema_2_1/src/Problema_2_1.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dado como dato el número de sonidos emitidos por un
 * grillo en un minuto, calcula la temperatura en grados Fahrenheit.
 *
 * @author Angel P.
 * 
 */

public class Problema_2_1 {

    public static float calcularTemperatura(int noSonidos) {

        return (float) noSonidos / 4 + 40;
        // Función de temperatura en grados Fahrenheit
        // de acuerdo al no. de sonidos emitidos por un grillo en un minuto
    }

    public static void main(String[] args) {

        int noSonidos;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.println("Sonidos por minuto de un grillo: ");
            noSonidos = scanner.nextInt();

            if (noSonidos <= 0) {

                System.out.println("Error: el no. de sonidos debe ser mayor que cero.");
                return;
            }
        } catch (InputMismatchException e) {

            System.out.println("Error: por favor, ingresa un número entero.");
            return;
        }
        float temperatura = calcularTemperatura(noSonidos);

        System.out.println("La temperatura es igual a");

        System.out.printf("%.2f grados Fahrenheit", temperatura);
    }
}